content/blog/deploy-kubernetes-and-apps-with-go/index.md

```diff
@@ -0,0 +1,309 @@
+date: 2020-04-14
+meta_desc: "Build Kubernetes deployments with Go using Pulumi's reusable components."
+meta_image: gopher.png
+authors:
+    - levi-blackstone
+tags:
+    - Kubernetes
+    - Go
+    - containers
+---
+
+We're excited to announce that Go as a first-class language in Pulumi, and you can now build your infrastructure with Go on AWS, Azure, GCP, and many other clouds. Users often ask — "can I use Pulumi to manage Kubernetes infrastructure in Go today?" With the upcoming release of Pulumi 2.0., the answer is "yes!"
+
+<!--more-->
+
+Building your Kubernetes infrastructure with Infrastructure as Code offers several benefits:
+
+- Strongly-typed inputs and outputs for resources and invokes
+- First-class language support in editors/IDEs like vim, VS Code, GoLand, and atom
+- Reusable components and classes: Abstract standard functionality into classes and libraries for code reuse, and clean infrastructure design, instead of copy/pasting pages of YAML.
+- Embed Pulumi programs within other Go-based tools
+
+These benefits provide a productive experience for Go developers using Kubernetes.
+
+## Tour of Kubernetes with Go
+
+Let's start with a basic example, e.g.,  deploying an Nginx pod into a Kubernetes cluster.
+
+```go
+pod, err := corev1.NewPod(ctx, "pod", &corev1.PodArgs{
+	Spec: corev1.PodSpecArgs{
+		Containers: corev1.ContainerArray{
+			corev1.ContainerArgs{
+				Name:  pulumi.String("nginx"),
+				Image: pulumi.String("nginx"),
+			}
+		},
+	},
+})
+if err != nil {
+		return err
+}
+```
+
+We can deploy this with `pulumi up`.
+
+```
+$ pulumi up
+Previewing update (nginx):
+     Type                    Name                 Plan
+ +   pulumi:pulumi:Stack     kubernetes-go-nginx  create
+ +   └─ kubernetes:core:Pod  pod                  create
+
+Resources:
+    + 2 to create
+
+Do you want to perform this update? yes
+Updating (nginx):
+     Type                    Name                 Status
+ +   pulumi:pulumi:Stack     kubernetes-go-nginx  created
+ +   └─ kubernetes:core:Pod  pod                  created
+
+Resources:
+    + 2 created
+
+Duration: 16s
+```
+
+Although programming languages are imperative, Pulumi describes the desired state of the infrastructure. If we change our program, Pulumi will compute the minimum delta and apply it to our Kubernetes cluster, which will transition to the new desired state. For example, we can modify our Kubernetes resources in place inside the cluster by adding a label to our Pod:
+
+```patch
+pod, err := corev1.NewPod(ctx, "pod", &corev1.PodArgs{
++	Metadata: &metav1.ObjectMetaArgs{
++		Labels: pulumi.StringMap{"app": pulumi.String("nginx")},
++	},
+	Spec: corev1.PodSpecArgs{
+		Containers: corev1.ContainerArray{
+			corev1.ContainerArgs{
+				Name:  pulumi.String("nginx"),
+				Image: pulumi.String("nginx"),
+			}},
+	},
+})
+if err != nil {
+	return err
+}
+```
+
+The Pulumi preview shows that the pod is updated in place with the new label.
```

Comment:
add pops for pulumi preview instead?

Reply:
made it more explicit: 

When we update the deployment with 'pulumi up`, the preview shows that the pod is updated in place with the new label.